### PR TITLE
Add integration tests for nxos_acl_interface

### PR DESCRIPTION
**Pull request by mikewiebe**
##### SUMMARY
This update
  * Adds integration tests for the `nxos_acl_interface` module.
  * Adds dynamic interface selection and platform gathering info in `test/integration/targets/prepare_nxos_tests/tasks/main.yml`  that can be used in any of the integration tests.

##### ISSUE TYPE
 - Integration Test Pull Request

##### COMPONENT NAME
nxos_acl_interface

##### ANSIBLE VERSION
```
ansible 2.4.0 (it/nxos_acl_interface 4c707a3c55) last updated 2017/06/27 10:17:45 (GMT -400)
  config file = None
  configured module search path = [u'/Users/mwiebe/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mwiebe/Projects/nxos_ansible/fix_ansible/lib/ansible
  executable location = /Users/mwiebe/Projects/nxos_ansible/fix_ansible/bin/ansible
  python version = 2.7.13 (default, Apr  4 2017, 08:47:57) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]
```

### CODE REVIEW

File: test/integration/targets/prepare_nxos_tests/tasks/main.yml [hunk 1]
```diff
@@ -1,9 +1,55 @@
+---
 # NXAPI is enabled differently on NX-OS platforms.
 # Try both commands to enable NXAPI and ignore any errors.
 - name: enable nxapi on remote device
   nxos_config:
     lines:
-      - feature cli
```
Comment (mikewiebe) [inline test/integration/targets/prepare_nxos_tests/tasks/main.yml:6]: `feature cli` and `feature cli sandbox` were incorrectly used.  `nxapi` should be used instead.

File: test/integration/targets/prepare_nxos_tests/tasks/main.yml [hunk 2]
```diff
@@ -1,9 +1,55 @@
...
-      - feature cli sandbox
+      - feature nxapi
+      - feature nxapi sandbox
```
Comment (Qalthos) [inline test/integration/targets/prepare_nxos_tests/tasks/main.yml]: I can't find a reference to `feature nxapi sandbox` being a valid command on any platform. Did you mean `nxapi sandbox` or is there something I'm missing here?

Comment (mikewiebe) [reply test/integration/targets/prepare_nxos_tests/tasks/main.yml]: @Qalthos Yes, good catch. It should be `nxapi sandbox`.  I will update the diffs.

Comment (mikewiebe) [reply test/integration/targets/prepare_nxos_tests/tasks/main.yml]: Thinking more however.. I don't think that we need the sandbox for Ansible testing.  I believe just `feature nxapi` should be sufficient.  I will confirm and then remove `sandbox` if possible.

**Pull request merged**